lib/fizzy/execute.cpp

```diff
@@ -1437,4 +1448,85 @@ std::optional<FuncIdx> find_exported_function(const Module& module, std::string_
 
     return {};
 }
+
+std::optional<ExternalFunction> find_exported_function(Instance& instance, std::string_view name)
+{
+    const auto opt_index = find_export(instance, ExternalKind::Function, name);
+    if (!opt_index.has_value())
+        return std::nullopt;
+
+    return [idx = *opt_index, &instance](fizzy::Instance&, std::vector<uint64_t> args) {
+        return execute(instance, idx, std::move(args));
+    };
+}
+
+std::optional<ExternalGlobal> find_exported_global(Instance& instance, std::string_view name)
+{
+    const auto opt_index = find_export(instance, ExternalKind::Global, name);
+    if (!opt_index.has_value())
+        return std::nullopt;
+
+    const auto global_idx = *opt_index;
+    if (global_idx < instance.imported_globals.size())
+    {
+        // imported global is reexported
+        return ExternalGlobal{instance.imported_globals[global_idx].value,
+            instance.imported_globals[global_idx].is_mutable};
+    }
+    else
+    {
+        // global owned by instance
+        return ExternalGlobal{
+            &instance.globals[global_idx], instance.module.globalsec[global_idx].is_mutable};
```

Comment:
Index of the global is incorrect here, should be `global_idx - instance.imported_globals.size()`

Reply:
Fixed, also modified the test to contain both imported and non-imported globals.